plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,276 @@
+	case "application/json":
+		body, _ := ioutil.ReadAll(resp.Body)
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		err = gatherStatusUrl(bufio.NewReader(resp.Body), n.getTags(addr), acc)
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		err = gatherWeatherUrl(bufio.NewReader(resp.Body), n.getTags(addr), acc)
+		return err
+	default:
+		return fmt.Errorf("%s returned unexpected content type %s", addr.String(), contentType)
+	}
+}
+
+func (n *OpenWeatherMap) getTags(addr *url.URL) map[string]string {
+	h := addr.Host
+	host, port, err := net.SplitHostPort(h)
+	if err != nil {
+		host = addr.Host
+		if addr.Scheme == "http" {
+			port = "80"
+		} else if addr.Scheme == "https" {
+			port = "443"
+		} else {
+			port = ""
+		}
+	}
+	tags := map[string]string{
+		"server":   host,
+		"port":     port,
+		"base_url": n.BaseUrl,
+	}
+	return tags
+}
+
+type WeatherEntry struct {
+	Dt     int64  `json:"dt"`
+	Dttxt  string `json:"dt_txt"` // empty for weather/
+	Clouds struct {
+		All int64 `json:"all"`
+	} `json:"clouds"`
+	Main struct {
+		GrndLevel float64 `json:"grnd_level"` // empty for weather/
+		Humidity  int64   `json:"humidity"`
+		SeaLevel  float64 `json:"sea_level"` // empty for weather/
+		Pressure  float64 `json:"pressure"`
+		Temp      float64 `json:"temp"`
+		TempMax   float64 `json:"temp_max"`
+		TempMin   float64 `json:"temp_min"`
+	} `json:"main"`
+	Rain struct {
+		Rain3 float64 `json:"3h"`
+	} `json:"rain"`
+	Sys struct {
+		Pod     string  `json:"pod"`
+		Country string  `json:"country"`
+		Message float64 `json:"message"`
+		Id      int64   `json:"id"`
+		Type    int64   `json:"type"`
+		Sunrise int64   `json:"sunrise"`
+		Sunset  int64   `json:"sunset"`
+	} `json:"sys"`
+	Wind struct {
+		Deg   float64 `json:"deg"`
+		Speed float64 `json:"speed"`
+	} `json:"wind"`
+	Weather []struct {
+		Description string `json:"description"`
+		Icon        string `json:"icon"`
+		Id          int64  `json:"id"`
+		Main        string `json:"main"`
+	} `json:"weather"`
+
+	// Additional entries for weather/
+	Id    int64  `json:"id"`
+	Name  string `json:"name"`
+	Coord struct {
+		Lat float64 `json:"lat"`
+		Lon float64 `json:"lon"`
+	} `json:"coord"`
+	Visibility int64 `json:"visibility"`
+}
+
+type Status struct {
+	City struct {
+		Coord struct {
+			Lat float64 `json:"lat"`
+			Lon float64 `json:"lon"`
+		} `json:"coord"`
+		Country string `json:"country"`
+		Id      int64  `json:"id"`
+		Name    string `json:"name"`
+	} `json:"city"`
+	List []WeatherEntry `json:"list"`
+}
+
+func gatherStatusUrl(r *bufio.Reader, tags map[string]string, acc telegraf.Accumulator) error {
+	dec := json.NewDecoder(r)
+	status := &Status{}
+	if err := dec.Decode(status); err != nil {
+		return fmt.Errorf("Error while decoding JSON response: %s", err)
+	}
+	status.Gather(tags, acc)
+	return nil
+}
+
+func gatherWeatherUrl(r *bufio.Reader, tags map[string]string, acc telegraf.Accumulator) error {
+	dec := json.NewDecoder(r)
+	status := &Status{}
```

Comment:
Making the current weather part of a forecast in order to process feels wrong. Consider revising.

Reply:
Both JSON structures given by the OWM API are nearly the same (weather info structure in the past, present, future), that's why I'm reusing the code to parse the two API responses.